### DISCUSSION TITLE
Is there advanced/detailed logging available?

### DISCUSSION THREAD

**Discussion by pborah88** [Stability-AI/StableSwarmUI#303] [Q&A]
First of all, thank you for answering my silly questions so far. I sincerely appreciate it. I come from a tinkerer background and not a hardcore programmer. I only got into this whole AI image generation thing less than a month ago. Up until a week ago or so I was using ComfyUI before I found out about StableSwarmUI on the SD reddit sub. Now, onto my question.

Is it possible to track advanced/detailed logging for StableSwarm?

When generating multiple images using advanced parameters like refiner, SAG, ClipSkip, FreeU, etc. the UI freezes mid generation and nothing short of closing and restarting works. I've tried restarting individual backends, all together, free up VRAM or system RAM, it does not help/work. The interrupts work sometimes if it's early enough into the freeze otherwise they don't. The console doesn't show anything with regards to  at what step there may have been some error. The one thing I've noticed is my GPU memory is usually at 20%-50% usage when it freezes and doesn't budge even when I click Free VRAM.

Since I'm using this on a test bench PC, I can physically see the GPU fans no longer running and touching the GPU backplate feels cool so I know there's no processing going on anymore. I only have 16 GB of system RAM and it's usually maxing out during generations, could this be a possible issue? I have two GPUs with 8 GB VRAM each.

My usual parameter configs are widescreen (1280x720ish) generations with an XL model, dpm++ karras, 20-30 steps and 3-8 cfg. a small XL refiner with 20-40% refiner control and about 10-20 refiner steps. Upscaling using a separate upscale model, SAG usually between 1-1.5 and blur of 0.5-0.75, ocassionally I use clipskip and FreeU as well. Thanks.

**Comment by mcmonkey4eva** [comment]
Yes, go to Server -> Logs -> set "view type" to "Debug" or even "Verbose"
You can also go to Server Configuration to change LogLevel for what shows up in the console view.
It's like the Debug view will show you what's wrong, as that contains raw output from the Comfy backend.
